src/control/lib/control/auto_test.go

```diff
@@ -998,6 +998,12 @@ func TestControl_AutoConfig_genConfig(t *testing.T) {
 					config.Server{},
 					sysfs.Provider{},
 				),
+				cmp.Comparer(func(x, y *storage.BdevDeviceList) bool {
+					if x == nil && y == nil {
+						return true
+					}
+					return x.Equals(y)
```

Comment:
Could be usefull to also test if x == nil || y == nil. Otherwise, the return statement could crash if x is nil and not y.

Reply:
The implementation of [BdevDeviceList.Equals()](https://github.com/daos-stack/daos/pull/8032/files#diff-af190e716da9a8067f6406eb4291dc08a2afabe0de9cb23599f29651e5d98cf8R387) handles nil `x` or `y` already.